torchaudio/models/wav2vec2/model.py

```diff
@@ -116,6 +116,125 @@ def forward(
         return x, lengths
 
 
+class HuBERTModel(Wav2Vec2Model):
+    """HuBERT models for training from scratch.
+
+    Note:
+        To build the model, please use one of the factory functions.
+
+    Args:
+        feature_extractor (torch.nn.Module):
+            Feature extractor that extracts feature vectors from raw audio Tensor.
+
+        encoder (torch.nn.Module):
+            Encoder that converts the audio features into the sequence of probability
+            distribution (in negative log-likelihood) over labels.
+
+        mask_generator (torch.nn.Module):
+            Generate the mask for masked prediction during the training.
+
+        mask_embedding (torch.nn.Parameter):
+
+        label_embeddings (torch.nn.Parameter):
+
+        final_proj (torch.nn.Module):
+    """  # noqa: E501
+    def __init__(
+        self,
+        feature_extractor: Module,
+        encoder: Module,
+        mask_generator: Module,
+        mask_embedding: Parameter,
+        label_embeddings: Parameter,
+        final_proj: Module,
+    ):
+        super().__init__(feature_extractor, encoder)
+        self.feature_extractor = feature_extractor
+        self.encoder = encoder
+        self.mask_generator = mask_generator
+        self.mask_embedding = mask_embedding
```

Comment:
I am aware that with the use of factory functions, users do not need to build `Parameter` by themselves, but it appears somewhat unconventional to have `Parameter` as a constructor arguments.

Looking at how `mask_embedding` is used, it is only passed to `mask_generator`. Can't `mask_embedding` be part of `mask_generator`?

Reply:
Sure, let me add ``mask_embedding`` into ``mask_generator``, but for ``label_embeddings``, maybe I can create another module class named ``LogitsGenerator`` to include it?